client/my-sites/site-settings/spam-filtering-settings.jsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { includes } from 'lodash';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import FoldableCard from 'components/foldable-card';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import FormInputValidation from 'components/forms/form-input-validation';
+import InfoPopover from 'components/info-popover';
+import ExternalLink from 'components/external-link';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getJetpackSettingsSaveError, getJetpackSettingsSaveRequestStatus } from 'state/selectors';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+
+const SpamFilteringSettings = ( {
+	fields,
+	dirtyFields,
+	hasAkismetKeyError,
+	isRequestingSettings,
+	isSavingSettings,
+	onChangeField,
+	translate
+} ) => {
+	const { wordpress_api_key, akismet: akismetActive } = fields;
+	const isDirty = includes( dirtyFields, 'wordpress_api_key' );
+	const header = ( akismetActive && wordpress_api_key && ! isDirty )
```

Comment:
> character in the API key field, then delete it to return to the previous state, it will not display that it's valid, unless you refresh.

I think this is fine. It represents the logic of a field being dirty until you save it. 

> The same happens if you have an empty field or a wrong API key, and then you submit with the valid API key - you will still see the "Your site needs an Antispam key." message, and the field validation will not be

I'll check this, this is wrong. Thank you